tests/unit/worker/task/test_catalog_cleanup.py

```diff
@@ -144,6 +162,11 @@ async def test_catalog_cleanup(mock_drop_table, catalog_with_associated_document
     catalog_cleanup_task = app_container.catalog_cleanup_task
     catalog_service = app_container.catalog_service
     await catalog_service.soft_delete_document(document_id=catalog.id)
+
+    # add a delay to ensure the soft delete is completed
+    await asyncio.sleep(1)
```

Comment:
Is this sleep necessary? My understanding is that the `soft_delete_document` above is done synchronously.

Reply:
When i run this locally (for this test), it is quite flaky. If I run the whole unit tests, it passes without any issue. After adding this delay, the issue gone.